src/main/java/org/minions/devfund/noemiguzman/battleship/Ship.java

```diff
@@ -0,0 +1,178 @@
+package org.minions.devfund.noemiguzman.battleship;
+
+/**
+ * Ship class.
+ */
+public abstract class Ship {
+    private int length;
+    private int bowRow;
+    private int bowColumn;
+    private boolean horizontal;
+    private boolean[] hit;
+
+    /**
+     * This is an abstract method and therefore has no body.
+     *
+     * @return ship string
+     */
+    public abstract String getShipType();
+
+    /**
+     * Returns true if it is okay to put a ship of this length with its bow in this location,
+     * with the given orientation, and returns false otherwise.
+     * The ship must not overlap another ship, or touch another ship (vertically, horizontally, or diagonally),
+     * and it must not stickout beyond the array.
+     * Do not actually change either the ship or the Ocean, just says whether it is legal to do so.
+     *
+     * @param row        numeric
+     * @param column     numeric
+     * @param horizontal true
+     * @param ocean      object
+     * @return true
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+
+        final int oceanLength = ocean.getShipArray().length;
+        if (horizontal) {
+            if (column + length > oceanLength) {
+                return false;
+            }
+            return helpToPlaceShipAt(column, row, 1, length + 1, ocean);
+
+        } else {
+            if (row + length > oceanLength) {
+                return false;
+            }
+            return helpToPlaceShipAt(column, row, length, 1, ocean);
+        }
+    }
+
+    /**
+     * help to ask to place ship at.
+     *
+     * @param column       numeric
+     * @param row          numeric
+     * @param lenghtRow    maxrow
+     * @param lengthColumn maxcolumn
+     * @param ocean        object
+     * @return true
+     */
+    private boolean helpToPlaceShipAt(int column, int row, int lenghtRow, int lengthColumn, final Ocean ocean) {
+
+        for (int i = row - 1; i <= row + lenghtRow; i++) {
+            for (int j = column - 1; j < column + lengthColumn; j++) {
+                try {
+
+                    if (!ocean.getShipArray()[i][j].getShipType().equals("empty")) {
+                        return false;
+                    }
+                } catch (Exception e) {
```

Comment:
Please don't use try catch in this scenario
You should validate first the positions.